src/JustSaying/AwsTools/MessageHandling/SnsPolicy.cs

```diff
@@ -1,56 +1,58 @@
-using System.Collections.Generic;
-using System.Linq;
+using System;
+using System.Text.Json;
 using System.Text.RegularExpressions;
 using System.Threading.Tasks;
-using Amazon.Auth.AccessControlPolicy;
-using Amazon.Auth.AccessControlPolicy.ActionIdentifiers;
 using Amazon.SimpleNotificationService;
 using Amazon.SimpleNotificationService.Model;
 
 namespace JustSaying.AwsTools.MessageHandling
 {
-    public class SnsPolicy
+    internal static class SnsPolicy
     {
-        private readonly IReadOnlyCollection<string> _accountIds;
-
-        public SnsPolicy(IReadOnlyCollection<string> accountIds)
-        {
-            _accountIds = accountIds;
-        }
-
-        public async Task SaveAsync(string sourceArn, IAmazonSimpleNotificationService client)
+        internal static async Task SaveAsync(SnsPolicyDetails policyDetails, IAmazonSimpleNotificationService client)
         {
-            ActionIdentifier[] actions = { SNSActionIdentifiers.Subscribe };
-
-            var snsPolicy = new Policy()
-                .WithStatements(GetDefaultStatement(sourceArn))
-                .WithStatements(new Statement(Statement.StatementEffect.Allow)
-                    .WithPrincipals(_accountIds.Select(a => new Principal(a)).ToArray())
-                    .WithResources(new Resource(sourceArn))
-                    .WithActionIdentifiers(actions));
-            var attributeValue = snsPolicy.ToJson();
-            var setQueueAttributesRequest = new SetTopicAttributesRequest(sourceArn, "Policy", attributeValue);
+            var sourceAccountId = ExtractSourceAccountId(policyDetails.SourceArn);
```

Comment:
Is switching from the fluent api to json here because the fluent api is out of date from the SQS API?

Reply:
Looks like its from the SDK - did they just remove it?

Reply:
It's been marked as obsolete, and the message says to use JSON going forwards

Reply:
It's not broken, but they say they won't be keeping it up to date

Reply:
Ah that's interesting, fair enough. I wonder if we should keep this json in an external file as a template that we load on startup. Likely something we can put off until we do the work to disable creation on startup, as we'd want to remove the permissions for those actions too.

Reply:
Trying to ship a template as an external file via NuGet would be a bit bleurgh - we could embed it as a JSON resource, but it still wouldn't be user editable. Would be good to provide a way to override the policy and let the user provide an entire custom one instead.

Reply:
I was thinking as an embedded file, yeah. If we wanted it to be user editable I guess we could provide well-known settings that we substitute in where it makes sense.